src/main/java/de/digitalcollections/solrocr/formats/alto/AltoPassageFormatter.java

```diff
@@ -122,6 +125,49 @@ protected String getTextFromXml(String altoFragment) {
         .replaceAll(END_HL, endHlTag);
   }
 
+  @Override
+  protected String getHighlightedFragment(Passage passage, IterableCharSequence content) {
+    StringBuilder sb = new StringBuilder(content.subSequence(passage.getStartOffset(), passage.getEndOffset()));
+    int extraChars = 0;
+    if (passage.getNumMatches() > 0) {
+      List<PassageMatch> matches = mergeMatches(passage.getNumMatches(), passage.getMatchStarts(), passage.getMatchEnds());
+      for (PassageMatch match : matches) {
+        String preMatchContent = content.subSequence(passage.getStartOffset(), match.start).toString();
+        int matchStart = preMatchContent.length();
+        if (alignSpans) {
+          matchStart = preMatchContent.lastIndexOf("CONTENT=") + 9;
+        }
+        sb.insert(matchStart + extraChars, startHlTag);
+        extraChars += startHlTag.length();
+        int matchEnd = content.subSequence(passage.getStartOffset(), match.end).toString().length();
+        String matchText = sb.substring(extraChars + matchStart, extraChars + matchEnd);
+        if (matchText.trim().endsWith(">")) {
+          // Set the end of the match to the position before the last inner closing tag inside of the match.
+          Matcher m = LAST_INNER_TAG_PAT.matcher(matchText);
+          int idx = -1;
+          while (m.find()) {
+            idx = m.start() + 1;
+          }
+          if (idx > -1) {
+            matchEnd -= (matchText.length() - idx);
+          }
+        }
```

Comment:
Ist der Fall im Test schon abgedeckt?

Reply:
Für den hOCR Fall, ja, in `HocrTest#testMaskedDocumentIsIndexed`. Mit ALTO allerdings noch nicht. Mir fällt allerdings gerade auf dass der Fall in ALTO eigentlich garnicht auftreten kann, da der Match sich ja immer innerhalb von einem oder mehreren `CONTENT="..."` Blöcken abspielt, der Fall dass ein Match mit einem Element endet tritt eigentlich nur bei hOCR auf. Ich werde den Branch daher entfernen, das sollte den Code etwas einfacher machen. Danke für's hinterfragen :-)